pkg/server/container_create.go

```diff
@@ -192,44 +192,49 @@ func (c *criContainerdService) generateContainerSpec(id string, sandboxPid uint3
 		g.AddProcessEnv(e.GetKey(), e.GetValue())
 	}
 
-	// TODO: add setOCIPrivileged group all privileged logic together
 	securityContext := config.GetLinux().GetSecurityContext()
 
 	// Add extra mounts first so that CRI specified mounts can override.
-	addOCIBindMounts(&g, append(extraMounts, config.GetMounts()...), securityContext.GetPrivileged())
+	addOCIBindMounts(&g, append(extraMounts, config.GetMounts()...))
 
-	g.SetRootReadonly(securityContext.GetReadonlyRootfs())
+	if securityContext.GetPrivileged() {
+		if err := setOCIPrivileged(&g, config); err != nil {
+			return nil, err
+		}
+	} else {
+		if err := addOCIDevicesWithoutPrivilege(&g, config.GetDevices()); err != nil {
```

Comment:
Please follow the same pattern. For the other functions, you are using `setOCICapabilities` vs SetupPrivileged, `addOCIBindMounts` vs `setOCIBindMountsPrivileged`.

How about we use `addOCIDevices` vs `setOCIDevicesPrivileged` to make them consistent?